README.md

```diff
@@ -1,41 +1,108 @@
-# Open Source Project Template
+# GeolocatorService
 
-This repository contains a template to seed a repository for an Open Source
-project.
+A service to get the user's current location, based on Geolocator, suited to UWP or [Uno](https://platform.uno/) apps.
 
-## How to use this template
+GeolocatorService aims to simplify getting the user's current location and handling most common scenarios, including getting the permission to obtain this location and handling cases where this permission is denied.
 
-1. Check out this repository
-2. Delete the `.git` folder
-3. Git init this repository and start working on your project!
-4. Prior to submitting your request for publication, make sure to review the
-   [Open Source guidelines for publications](https://nventive.visualstudio.com/Internal/_wiki/wikis/Internal_wiki?wikiVersion=GBwikiMaster&pagePath=%2FOpen%20Source%2FPublishing&pageId=7120).
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](LICENSE)
 
-## Features (to keep as-is, configure or remove)
-- [Mergify](https://mergify.io/) is configured. You can edit or remove [.mergify.yml](/.mergify.yml).
-- [allcontributors](https://allcontributors.org/) is configured. It helps adding contributors to the README.
-- [dependabot](https://dependabot.com/) is configured. This bot will open pull requests automatically to update nuget dependencies. This one could be annoying, feel free to remove the [.dependabot](/.dependabot) folder.
+## Getting Started
 
-The following is the template for the final README.md file:
+Add the GeolocatorService nuget package to your project heads.
 
----
+To have access to reactive extensions such as ```GetAndObserveLocation```, add the GeolocatorService.Reactive nuget package to your project.
 
-# Project Title
+### Add the relevant permissions
 
-{Project tag line}
+#### UWP
 
-{Small description of the purpose of the project}
+Add the Location capability in your manifest.
 
-[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](LICENSE)
+#### iOS
 
-## Getting Started
+Add the NSLocationWhenInUsageDescription and NSLocationUsageDescription values to your info.plist file. This is the message that is displayed to your user when the permission for their location is requested. For instance,
+
+```
+<key>NSLocationWhenInUsageDescription</key>
+<string>TODO NSLocationWhenInUsageDescription</string>
+<key>NSLocationUsageDescription</key>
+<string>TODO NSLocationUsageDescription</string>
+```
+
+#### Android
+
+Add the location permission to your AssemblyInfo.cs file.
+
+```
+[assembly: UsesPermission(Android.Manifest.Permission.AccessFineLocation)]
+```
+
+### Request permission
+
+After you have added the relevant location permission info to your project configuration, locate the spot in your code where you want to ask the user for their location permission. This must be done before accessing their location. Add the following line to do so:
```

Comment:
We could give this reference to help the dev decide "when" https://developer.apple.com/design/human-interface-guidelines/ios/app-architecture/requesting-permission/

Reply:
and this https://developer.android.com/training/permissions/requesting

And/Or we can say "Ask permissions in context, in order for your users to intuitively know why the permission was requested."